### PR TITLE
build: enhance metadata json output

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/moby/buildkit/pull/2610

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: commands/build.go [hunk 1]
```diff
@@ -496,6 +493,32 @@ func parseContextNames(values []string) (map[string]string, error) {
...
 
+func writeMetadataFile(filename string, dt interface{}) error {
+	b, err := json.Marshal(dt)
+	if err != nil {
+		return err
+	}
+	return ioutils.AtomicWriteFile(filename, b, 0644)
+}
+
+func decExporterResponse(exporterResponse map[string]string) map[string]interface{} {
```
Comment (tonistiigi) [inline commands/build.go:504]: nit: `decodeExporterResponse` .

File: hack/test-driver [hunk 2]
```diff
@@ -44,58 +51,111 @@ if [ "$DRIVER" != "docker" ]; then
...
 
 # build
-${BUILDX_CMD} build ${buildPlatformFlag} \
-  --output="type=cacheonly" \
+buildxCmd build ${buildPlatformFlag} \
+  --output="$(buildOutput buildx-test-build)" \
   --builder="${builderName}" \
+  --metadata-file="${context}/metadata-build.json" \
   "${context}"
+jq '.' "${context}/metadata-build.json"
```
Comment (tonistiigi) [inline hack/test-driver:125]: Shouldn't this try some actual key?

Comment (crazy-max) [reply hack/test-driver:125]: I'm using `.` filter just to pretty print: https://github.com/docker/buildx/runs/5139711248?check_suite_focus=true#step:9:143
No validation there. Maybe we could add some though.

File: commands/build.go [hunk 3]
```diff
@@ -496,6 +493,32 @@ func parseContextNames(values []string) (map[string]string, error) {
 	return result, nil
 }
 
+func writeMetadataFile(filename string, dt interface{}) error {
+	b, err := json.Marshal(dt)
```
Comment (tonistiigi) [inline commands/build.go:497]: I guess removing `Marshalndent` was not intentional?

Comment (crazy-max) [reply commands/build.go:497]: Yes it was. I should have commented about that sorry. I think we should have the same behavior as `buildctl` but can revert that change if you want to.

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**